src/smt/options_manager.cpp

```diff
@@ -0,0 +1,148 @@
+               << std::endl;
+  if (key == options::defaultExprDepth.getName())
+  {
+    int depth = (*d_options)[options::defaultExprDepth];
+    Debug.getStream() << expr::ExprSetDepth(depth);
+    Trace.getStream() << expr::ExprSetDepth(depth);
+    Notice.getStream() << expr::ExprSetDepth(depth);
+    Chat.getStream() << expr::ExprSetDepth(depth);
+    Message.getStream() << expr::ExprSetDepth(depth);
+    Warning.getStream() << expr::ExprSetDepth(depth);
+    // intentionally exclude Dump stream from this list
+  }
+  else if (key == options::defaultDagThresh.getName())
+  {
+    int dag = (*d_options)[options::defaultDagThresh];
+    Debug.getStream() << expr::ExprDag(dag);
+    Trace.getStream() << expr::ExprDag(dag);
+    Notice.getStream() << expr::ExprDag(dag);
+    Chat.getStream() << expr::ExprDag(dag);
+    Message.getStream() << expr::ExprDag(dag);
+    Warning.getStream() << expr::ExprDag(dag);
+    Dump.getStream() << expr::ExprDag(dag);
+  }
+  else if (key == options::printExprTypes.getName())
+  {
+    bool value = (*d_options)[options::printExprTypes];
+    Debug.getStream() << expr::ExprPrintTypes(value);
+    Trace.getStream() << expr::ExprPrintTypes(value);
+    Notice.getStream() << expr::ExprPrintTypes(value);
+    Chat.getStream() << expr::ExprPrintTypes(value);
+    Message.getStream() << expr::ExprPrintTypes(value);
+    Warning.getStream() << expr::ExprPrintTypes(value);
+    // intentionally exclude Dump stream from this list
+  }
+  else if (key == options::dumpModeString.getName())
+  {
+    const std::string& value = (*d_options)[options::dumpModeString];
+    Dump.setDumpFromString(value);
+  }
+  else if (key == options::printSuccess.getName())
+  {
+    bool value = (*d_options)[options::printSuccess];
+    Debug.getStream() << Command::printsuccess(value);
+    Trace.getStream() << Command::printsuccess(value);
+    Notice.getStream() << Command::printsuccess(value);
+    Chat.getStream() << Command::printsuccess(value);
+    Message.getStream() << Command::printsuccess(value);
+    Warning.getStream() << Command::printsuccess(value);
+    *options::out() << Command::printsuccess(value);
+  }
+  // otherwise, no action is necessary
+}
+
+void OptionsManager::finishInit(LogicInfo& logic, bool isInternalSubsolver)
+{
+  // set up the timeout
+  d_resourceManager->setHardLimit(options::hardLimit());
+  if ((*d_options)[options::perCallResourceLimit] != 0)
+  {
+    d_resourceManager->setResourceLimit(options::perCallResourceLimit(), false);
+  }
+  if ((*d_options)[options::cumulativeResourceLimit] != 0)
+  {
+    d_resourceManager->setResourceLimit(options::cumulativeResourceLimit(),
+                                        true);
+  }
+  if ((*d_options)[options::perCallMillisecondLimit] != 0)
+  {
+    d_resourceManager->setTimeLimit(options::perCallMillisecondLimit(), false);
+  }
+  if ((*d_options)[options::cumulativeMillisecondLimit] != 0)
```

Comment:
This conflicts with #4711 which removes cumulative time limits (from the resource manager).